### PR TITLE
Disable xcm method

### PR DESCRIPTION
**Pull request by zqhxuyuan**
issue: https://github.com/AcalaNetwork/Acala/issues/1557

### CODE REVIEW

**Review by shaunxw — COMMENTED** [review]
I would recommend to use allow list on `pallet-xcm`, instead of block list. Polkadot xcm pallet has been adding dispatchable calls in it's new releases, and it's easy to miss new added calls with block list.

**Comment by xlc** [comment]
best to do a exhaustive match listing all the calls so in future if new calls are introduced, it will be a compiling error

**Comment by zqhxuyuan** [comment]
As the Call enum has __Ignore case, so here the default case we return unimplemented!()

File: runtime/karura/src/lib.rs [hunk 1]
```diff
@@ -228,6 +215,28 @@ impl Contains<Call> for BaseCallFilter {
...
+					return false;
+				}
+				pallet_xcm::Call::force_xcm_version { .. }
+				| pallet_xcm::Call::force_default_xcm_version { .. }
+				| pallet_xcm::Call::force_subscribe_version_notify { .. }
+				| pallet_xcm::Call::force_unsubscribe_version_notify { .. } => {
+					return true;
+				}
+				_ => {
+					unimplemented!()
```
Comment (xlc) [inline runtime/karura/src/lib.rs:235]:
```suggestion
					unreachable!()
```

File: runtime/karura/src/lib.rs [hunk 2]
```diff
@@ -215,6 +215,28 @@ impl Contains<Call> for BaseCallFilter {
...
+				| pallet_xcm::Call::limited_teleport_assets { .. } => {
+					return false;
+				}
+				pallet_xcm::Call::force_xcm_version { .. }
+				| pallet_xcm::Call::force_default_xcm_version { .. }
+				| pallet_xcm::Call::force_subscribe_version_notify { .. }
+				| pallet_xcm::Call::force_unsubscribe_version_notify { .. } => {
+					return true;
+				}
+				_ => {
```
Comment (xlc) [inline runtime/karura/src/lib.rs:234]: Can you do this?

```suggestion
				__Ignored => {
```

Comment (zqhxuyuan) [reply runtime/karura/src/lib.rs:234]: yeh, I can fix this

**Review by xlc — APPROVED** [review]

**Comment by zqhxuyuan** [comment]
@zjb0807 would you look at the failed e2e-tests?

**Comment by zjb0807** [comment]
> @zjb0807 would you look at the failed e2e-tests?

Retry it works.

**Pull request merged**